src/app/data-model-interface/DataModel.h

```diff
@@ -55,22 +55,24 @@ class DataModel : public DataModelMetadataTree
     // event emitting, path marking and other operations
     virtual InteractionModelContext CurrentContext() const { return mContext; }
 
-    /// List reading has specific handling logic:
-    ///   `state` contains in/out data about the current list reading. MUST start with kInvalidListIndex on first call
-    ///
     /// Return codes:
-    ///   CHIP_ERROR_MORE_LIST_DATA_AVAILABLE (NOTE: new error defined for this purpose)
-    ///      - partial data written to the destination
-    ///      - destination will contain AT LEAST one valid list entry fully serialized
-    ///      - destination will be fully valid (it will be rolled back on partial list writes)
+    ///   CHIP_ERROR_ACCESS_DENIED:
```

Comment:
Sniffing specific error values like this is not reliable; something deep in the guts of some platform impl might return CHIP_ERROR_ACCESS_DENIED for some random reason, ReturnErrorOnFailure propagates it right out, and we have incorrect behavior.

Reply:
I could return a dedicated error (but that would map to access denied ... so not that great either) or return a variant of CHIP_ERROR or some different dedicated type to force mapping ... but that would just mean CHIP_ERROR or "special error" and just makes the interface harder.

I need a way to tell the caller what happend. If something deep in the guts of the platform says ACCESS_DENIED why is it not correct to treat it just like any other access denied? This would allow AAI to also say access defined and be handled correctly.

Reply:
>  If something deep in the guts of the platform says ACCESS_DENIED why is it not correct to treat it just like any other access denied? 

Because absolutely nothing guarantees that CHIP_ERROR_ACCESS_DENIED matches the "reading from the attribute in the path requires a privilege that is not granted to access the cluster in the path" concept from the spec.  Maybe it does, maybe it does not.  The spec doesn't say "ignore random denials" it says to ignore _very specific things_.

Concretely, just looking at our code various platform time implementations return CHIP_ERROR_ACCESS_DENIED from `ClockImpl::SetClock_RealTime`, say.  How do you guarantee that nothing in AAI somewhere tries to do that?  You can't.

Reply:
Updated code to keep mExpanded logic and notes about this being temporary updates and that code will change.